routers/UserSessionsRouter.py

```diff
@@ -130,5 +164,44 @@ def launch_learning(session_id: int, parameters: dict):
                 dataset_path = session.dataset_path
                 markup_path = session.data_markup_path
 
+        @router.post("/upload_dataset/{session_id}", status_code=status.HTTP_200_OK)
+        def upload_dataset(session_id: int, response: Response, file: UploadFile = File(...)):
+            try:
+                if (pathlib.Path(file.filename).suffix == '.rar') or (pathlib.Path(file.filename).suffix == '.zip'):
+                    path_to_dir = f'{os.getcwd()}\\dataset\\{session_id}'
+                    is_exist = os.path.exists(path_to_dir)
+                    if not is_exist:
+                        os.makedirs(path_to_dir)
```

Comment:
I will add both markup and dataset folders as volumes in docker so there is no need to make it manually